pot/address_test.go

```diff
@@ -0,0 +1,66 @@
+package pot
+
+import (
+	"testing"
+)
+
+type testAddress []byte
+
+func newTestAddress(a []byte) testAddress {
+	if a != nil {
+		return testAddress(a)
+	}
+	return testAddress(make([]byte, 32))
+}
+
+func (a testAddress) Address() []byte {
+	return []byte(a)
+}
+
+// TestDistance tests the correctness of the distance calculation
+// as well as the comparison method
+func TestDistance(t *testing.T) {
+	a := [32]byte{}
+	b := [32]byte{}
+	a[0] = 0x91
```

Comment:
I would really like to see table driven test with multiple address and expected distances, starting from trivial ones.

Also, why not just write address literals or their hex representation (like for distance) in this test to be explicit, instead to create an array, set one byte in it and call a special helper function with slice constructed from the array?